jcl/src/openj9.dataaccess/share/classes/com/ibm/dataaccess/ExternalDecimal.java

```diff
@@ -0,0 +1,172 @@
+/*[INCLUDE-IF DAA]*/
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2023
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+package com.ibm.dataaccess;
+
+public class ExternalDecimal {
+    /**
+     * Private constructor, class contains only static methods.
+     */
+    private ExternalDecimal() {
+        super();
+    }
+
+
+    private static byte EXTERNAL_DECIMAL_SPACE_SYMBOL = (byte)0x40;
+    private static int EXTERNAL_DECIMAL_INVALID_DIGIT = 2;
+    private static int EXTERNAL_DECIMAL_INVALID_SIGN = 1;
+
+    /**
+     * Checks the validity of a External Decimal, returns an integer indicating the status of the External Decimal.
+     *
+     * @param byteArray
+     *            source array.
+     * @param offset
+     *            starting offset of the External Decimal.
+     * @param precision
+     *            number of digits to be verified.
+     * @param decimalType
+     *            constant indicating the type of the decimal. Supporte values are
```

Comment:
typo here. Also as we are mentioning the constant from DecimalData, may be we should add that.